### PR TITLE
Add IntegerUnion

### PR DESCRIPTION
**Pull request by fingolfin**
See https://github.com/oscar-system/Oscar.jl/issues/679

CC @tthsqe12 @fieker @thofma

### CODE REVIEW

File: src/flint/fq.jl [hunk 1]
```diff
@@ -588,7 +588,7 @@ end
 ###############################################################################
 
 @doc Markdown.doc"""
-    FlintFiniteField(char::Union{fmpz, Integer}, deg::Int, s::AbstractString; cached = true)
+    FlintFiniteField(char::IntegerUnion, deg::Int, s::AbstractString; cached = true)
```
Comment (fingolfin) [inline src/flint/fq.jl:591]: Perhaps it should be left in docstrings as-is for clarity?

File: src/flint/FlintTypes.jl [hunk 2]
```diff
@@ -67,6 +67,8 @@ mutable struct fmpz <: RingElem
     fmpz(x::fmpz) = x
 end
 
+const IntegerUnion = Union{Integer,fmpz}
```
Comment (fingolfin) [inline src/flint/FlintTypes.jl:70]: Should there be a docstring here? If so, what should it say?

**Comment by wbhart** [comment]
Fantastic idea. I have missed this many times.

File: src/flint/FlintTypes.jl (hunk 2 above)
Comment (wbhart) [reply src/flint/FlintTypes.jl:70]: Of course this doesn't include Singular.ZZ, which is unfortunate. I don't see a way to do that cleanly, other than to have a union there also which includes this one, which is called something else (maybe just Singular.IntegerUnion not sure).

Anyhow, here it is enough to simply add something to the integers.md file at the top, stating that we provide a union of all Julia and Nemo integers called IntegerUnion.

Comment (fingolfin) [reply src/flint/FlintTypes.jl:70]: OK, I've added a little blurb, let me know what you think

Comment (wbhart) [reply src/flint/FlintTypes.jl:70]: Looks good. Thanks.

File: src/flint/fq.jl (hunk 1 above)
Comment (wbhart) [reply src/flint/fq.jl:591]: I'm fine with using the new IntegerUnion. See what @thofma says.

Comment (thofma) [reply src/flint/fq.jl:591]: Fine with me.

**Pull request merged**